core/src/test/java/org/apache/iceberg/util/FakeTicker.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.util;
+
+import com.github.benmanes.caffeine.cache.Ticker;
+import java.time.Duration;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicLong;
+
+/**
+ * A Ticker whose value can be advanced programmatically in test.
+ *
+ * This is modified from the Guava package com.google.common.testing.FakeTicker,
```

Comment:
Modified? How?

Reply:
I added these lines which take in a plain `Duration`: https://github.com/apache/iceberg/pull/3543/files/8f197962ca045e99bdcaacc5370854314ab361fa#diff-aab28f9cd9db09ced4aeb1b9ffab120458652bdb001c977a5c790677b4c76030R53-R56

It's not even core guava, but a guava test package that hasn't been updated in a long time. So if there's a way we can avoid pulling that in and provide proper attribution, that would be great.

Reply:
At this point actually, it's not even really a copy. There are several similar `FakeTickers` and I've removed all of the methods that came from Guava.

`read` is part of the interface and it has to be in nanos, hence the `AtomicLong` of nanos.

We might be able to just remove the notice about Guava given that I've removed all of their signatures and the interface is defined by the caffeine library.

Reply:
I have actually removed all of the Guava code from this.

This is an implementation of the interface from `caffeine`, and there are no methods from the Guava class in it anymore.

I can remove that part at the top, as there are many `FakeTicker`s abounding on the internet. This one only uses nanos because that's what the interface uses. But I did not run across one that used a `Duration` as input.

So this is arguably not from Guava anymore (plus we do credit Guava already).

Up to you how to proceed.

Reply:
If it's completely replaced at this point, then let's just remove the comment about being from Guava.

Reply:
Removed.